### PR TITLE
Run unit tests on Android

### PR DESCRIPTION
**Pull request by jmalanen**
* Fix a bug in Android RunLoop
* Enable unit tests on Android
* Run unit tests on Firebase
* Fix the Android API breakage validator CI bot

### CODE REVIEW

File: platform/android/src/test/test_runner.cpp [hunk 1]
```diff
@@ -1,10 +1,78 @@
...
+    std::thread runnerThread(runner);
+
+    while (true) {
+        ALooper_pollAll(-1, &outFd, &outEvents, reinterpret_cast<void**>(&source));
+        if (source != nullptr) {
+            source->process(app, source);
+        }
+
+        if (!running && !done) {
+            mbgl::Log::Info(mbgl::Event::General, "TestRunner done");
```
Comment (alexshalamov) [inline platform/android/src/test/test_runner.cpp]: std::call_once instead of done flag?

File: platform/android/src/test/test_runner.cpp [hunk 2]
```diff
@@ -1,10 +1,78 @@
...
-int main(int argc, char *argv[]) {
-    if (!mbgl::android::initRuntime(argc, argv)) {
-        return 1;
+#include <unistd.h>
+#include <thread>
+
+using namespace mbgl;
+using namespace mbgl::android;
+
+bool running = true;
```
Comment (alexshalamov) [inline platform/android/src/test/test_runner.cpp]: std::atomic bool?

File: platform/android/src/test/test_runner.cpp (hunk 1 above)
Comment (jmalanen) [reply platform/android/src/test/test_runner.cpp]: Done

File: platform/android/src/test/test_runner.cpp (hunk 2 above)
Comment (jmalanen) [reply platform/android/src/test/test_runner.cpp]: Done

**Pull request merged**